src/cryptography/x509.py

```diff
@@ -29,6 +31,77 @@ def __init__(self, msg, parsed_version):
         self.parsed_version = parsed_version
 
 
+class UnknownAttribute(Exception):
+    pass
+
+
+class Attribute(object):
```

Comment:
So that's even worse :-) How about NameAttribute, or something like that,
that scopes it to names?

Alex

On Mon, Feb 2, 2015 at 2:45 PM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/x509.py
> <https://github.com/pyca/cryptography/pull/1645#discussion_r23966474>:
>
> > @@ -29,6 +31,77 @@ def __init__(self, msg, parsed_version):
> >          self.parsed_version = parsed_version
> >
> >
> > +class UnknownAttribute(Exception):
> > +    pass
> > +
> > +
> > +class Attribute(object):
>
> http://tools.ietf.org/html/rfc5280#section-4.1.2.4 is where this is from.
> The RFC calls it AttributeTypeAndValue. Suggestion? I don't have a better
> name than Attribute right now.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1645/files#r23966474>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084